src/init.c

```diff
@@ -720,6 +768,367 @@ static int hyper_remove_container(char *json, int length)
 	return ret;
 }
 
+/**
+ * converts a line to an array, splits the line in tokens with strtok using
+ * horizontal spaces (\n\t ) as delimiter 
+ *
+ * \param line to process
+ * \param[out] array_size array size
+ * \param max_num_tokens max number of tokens that array should contain, if -1
+ * then there is no limit
+ *
+ * \return an array on success, else NULL
+ */
+static char **hyper_line_to_array(char *line, size_t *array_size, int max_num_tokens)
+{
+	char **array = NULL;
+	size_t size = 12;
+
+	if (max_num_tokens == 0) {
+		return NULL;
+	}
+
+	array = (char **)calloc(size, sizeof(*array));
+	if (array == NULL) {
+		return NULL;
+	}
+
+	size_t i = 0;
+	char *delim = " \n\t";
+	char *tok = strtok(line, delim);
+	while (tok != NULL) {
+		array[i++] = tok;
+
+		if (i >= size-1) {
+			char **tmp = NULL;
+			size += size;
+			tmp = (char **)realloc(array, size * sizeof(*array));
+			if (tmp == NULL) {
+				/* try to reallocate once again */
+				tmp = (char **)realloc(array, size * sizeof(*array));
+			}
+			if (tmp == NULL) {
+				/* realloc failed, free memory and return NULL */
+				free(array);
+				return NULL;
+			}
+			array = tmp;
+		}
+
+		if (max_num_tokens > 0 && (--max_num_tokens) == 0) {
+			break;
+		}
+
+		tok = strtok(NULL, delim);
+	}
+
+	if (array_size != NULL) {
+		*array_size = i;
+	}
+
+	/* Add NULL terminator */
+	array[i] = NULL;
+
+	return array;
+}
+
+/**
+ * gets the index of PID, for example in the next line pid index is 1
+ *    UID PID PPID C STIME TTY TIME CMD
+ *
+ * \param line to process
+ *
+ * \return pid index on success, else -1
+ */
+static int hyper_get_pid_index(const char *line)
+{
+	char **array = NULL;
+	char *new_line = NULL;
+	int ret = -1;
+	const char* pid_str = "PID";
```

Comment:
I'm wondering if it might be better to:

- Require `psargs` to specify the output format such that the first field *must be* the `pid`.
- If no `psargs` are specified, make hyperstart specify the format in the same way:
  `ps -o pid,ppid,pgrp,session,tpgid,tty,comm -e` or something).

The advantage of doing that:

- You could remove `pid_index` from `hyper_ps_data`.
- It would be easy to reject the `PSCONTAINER` command if the first field wasn't purely numeric on every line.
- You wouldn't need this function.
- The code would be safer since what happens if `psargs` is just `--no-headers` at the moment?